src/definitions/pinghf.json

```diff
@@ -0,0 +1,465 @@
+{
+    "messages": {
+        "set": {
+            "set_range": {
+                "id": "2001",
+                "description": "Set the scan range for acoustic measurements.",
+                "payload": [
+                    {
+                        "name": "scan_start",
+                        "type": "u32",
+                        "description:": "The beginning of the scan range in mm from the transducer.",
+                        "units": "mm"
+                    },
+                    {
+                        "name": "scan_length",
+                        "type": "u32",
+                        "description": "The length of the scan range.",
+                        "units": "mm"
+                    }
+                ]
+            },
+            "set_speed_of_sound": {
+                "id": "2002",
+                "description": "Set the speed of sound used for distance calculations.",
+                "payload": [
+                    {
+                        "name": "speed_of_sound",
+                        "type": "u32",
+                        "description": "The speed of sound in the measurement medium. ~1,500,000 mm/s for water.",
+                        "units": "mm/s"
+                    }
+                ]
+            },
+            "set_ping_interval": {
+                "id": "2004",
+                "description": "The interval between acoustic measurements.",
+                "payload": [
+                    {
+                        "name": "ping_interval",
+                        "type": "u16",
+                        "description": "The interval between acoustic measurements.",
+                        "units": "ms"
+                    }
+                ]
+            },
+            "set_gain_setting": {
+                "id": "2005",
+                "description": "Set the current gain setting.",
+                "payload": [
+                    {
+                        "name": "gain_setting",
+                        "type": "u8",
+                        "description": "The current gain setting. The analog gain can be determined by requesting the analog_gain_levels message."
+                    }
+                ]
+            },
+            "set_ping_enable": {
+                "id": "2006",
+                "description": "Enable or disable acoustic measurements.",
+                "payload": [
+                    {
+                        "name": "ping_enabled",
+                        "type": "u8",
+                        "description": "0: Disable, 1: Enable."
+                    }
+                ]
+            },
+            "set_auto_gain": {
+                "id": "2007",
+                "description": "Enable or disable automatic gain adjustments.",
+                "payload": [
+                    {
+                        "name": "enabled",
+                        "type": "u8",
+                        "description": "0: Disable, 1: Enable."
```

Comment:
Maybe we could use a bitfield ? Each bit could enable or disable an automatic feature.

Reply:
We can consider it. I think the python/arduino apis are easier to build when it is a single parameter per message. We will have to add bitfield logic to those templates if we do this.